packages/zoe/src/contracts/multipoolAutoswap/getCurrentPrice.js

```diff
@@ -55,30 +64,129 @@ export const makeGetCurrentPrice = (isSecondary, isCentral, getPool) => {
    * @returns {Amount} The amount required to be paid in order to gain amountOut
    */
   const getInputForGivenOutput = (amountOut, brandIn) => {
-    const { brand: brandOut, value: outputValue } = amountOut;
+    const { brand: brandOut } = amountOut;
 
     if (isCentral(brandIn) && isSecondary(brandOut)) {
-      return getPool(brandOut).getCentralToSecondaryOutputPrice(outputValue);
+      return getPool(brandOut).getPriceGivenRequiredOutput(brandIn, amountOut)
+        .amountIn;
     }
 
     if (isSecondary(brandIn) && isCentral(brandOut)) {
-      return getPool(brandIn).getSecondaryToCentralOutputPrice(outputValue);
+      return getPool(brandIn).getPriceGivenRequiredOutput(brandIn, amountOut)
+        .amountIn;
     }
 
     if (isSecondary(brandIn) && isSecondary(brandOut)) {
       // We must do two consecutive calls to get the price: from
       // the brandIn to the central token, then from the central
       // token to the brandOut.
-      const centralTokenAmount = getPool(
+      const { amountIn: centralTokenAmount } = getPool(
+        brandIn,
+      ).getPriceGivenRequiredOutput(brandIn, amountOut);
+      return getPool(brandOut).getPriceGivenRequiredOutput(
+        centralBrand,
+        centralTokenAmount,
+      ).amountIn;
+    }
```

Comment:
Is this correct? I would have thought that we need to have the `centralToken` only on one side of the two calls, rather than not at all in the first call, and on both sides in the second call:

```suggestion
      ).getPriceGivenRequiredOutput(centralBrand, amountOut);
      return getPool(brandOut).getPriceGivenRequiredOutput(
        brandIn,
        centralTokenAmount,
      ).amountIn;
    }
```

Reply:
Can we confirm this with a targeted unit test?

Reply:
This erroneous version has been subsumed by the other version. I've added a test case in `test-multipoolAutoswap.js`.  The error was in a different place, but you were right that it needed a test.